src/components/Microsoft.DotNet.UpgradeAssistant.MSBuild/SdkCollection.cs

```diff
@@ -0,0 +1,85 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Construction;
+
+namespace Microsoft.DotNet.UpgradeAssistant.MSBuild
+{
+    public class SdkCollection : ICollection<string>
+    {
+        private readonly ProjectRootElement _projectRoot;
+
+        public SdkCollection(ProjectRootElement projectRoot)
+        {
+            _projectRoot = projectRoot;
+        }
+
+        public HashSet<string> GetSdks() => _projectRoot.Sdk.Split(';').ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+        public ICollection<string> GetSdkLists() => _projectRoot.Sdk.Split(';').ToList();
+
+        public bool Contains(string item)
+        {
+            return GetSdks().Contains(item, StringComparer.OrdinalIgnoreCase);
```

Comment:
Why the comparer again? The GetSdks is returning hashset that has the right comparer. You're actually using a LINQ API that will iterate over everything